src/ziggurat/metrics.clj

```diff
@@ -2,43 +2,60 @@
   (:require [clojure.tools.logging :as log])
   (:import (com.gojek.metrics.datadog DatadogReporter)
            (com.gojek.metrics.datadog.transport UdpTransport$Builder UdpTransport)
-           (java.util.concurrent TimeUnit)
-           (io.dropwizard.metrics5 MetricRegistry Meter MetricName Histogram)))
+           (io.dropwizard.metrics5 MetricRegistry Meter MetricName Histogram)
+           (java.util.concurrent TimeUnit)))
 
 (defonce ^:private group (atom nil))
 
 (defonce metrics-registry
   (MetricRegistry.))
 
 (defn mk-meter
-  [category metric]
-  (let [metric-name (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
-        tagged-metric (.tagged ^MetricName metric-name ^"[Ljava.lang.String;" (into-array String ["actor" @group]))]
-    (.meter ^MetricRegistry metrics-registry ^MetricName tagged-metric)))
+  ([category metric]
+   (mk-meter category metric nil))
+  ([category metric topic-entity-name]
+   (let [metric-name   (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
+         tags          (if (nil? topic-entity-name)
+                         ["actor" @group]
+                         ["actor" @group "topic_name" topic-entity-name])
+         tagged-metric (.tagged ^MetricName metric-name ^"[Ljava.lang.String;" (into-array String tags))]
```

Comment:
line 17-21 is being duplicated in `mk-meter` and `mk-histogram`. Can it be pulled out to a function or multiple functions?

Reply:
done, code is extracted to a function called (get-tags)